py-polars/polars/datatypes/classes.py

```diff
@@ -538,16 +538,30 @@ class Enum(DataType):
 
     categories: list[str]
 
-    def __init__(self, categories: list[str]):
+    def __init__(self, categories: Iterable[str]):
         """
         A fixed set categorical encoding of a set of strings.
 
         Parameters
         ----------
         categories
-            Categories in the dataset.
+            Valid categories in the dataset.
 
         """
+        from polars.utils.various import ordered_unique
+
+        categories = list(categories)
+        unique_categories, dupes = ordered_unique(categories, return_duplicates=True)
```

Comment:
The happy path should be fastest, so you could do something like:

```
if contains_duplicates(categories):
    ...  # compute the actual duplicates and raise an informative error
```

`contains_duplicates` can be faster than the ordered unique function, either simply `len(set(categories)) == len(categories)` or something [more specialized](https://stackoverflow.com/a/56761643).

And I don't think returning the number of duplicates is really that useful here. Just printing the first duplicate is fine. So you could do something like:

```python
def contains_duplicates(iterable):
    seen = []
    for x in iterable:
        if x in seen:
            return x  # Early exit and return the duplicate
        seen.append(x)
    return False
```

(type signature here is a bit concerning but just to convey the idea)